tests/compile-test.rs

```diff
@@ -106,7 +109,8 @@ fn run_ui_toml() {
             .parent()
             .unwrap()
             .display()
-            .to_string(),
+            .to_string()
+            .replace('\\', "/"),
```

Comment:
I think this'd still need to go through `regex::escape`, as canonical windows paths contain `?`, e.g. `\\?\D:\Code\Rust\clippy\`